config/initializers/flipper.rb

```diff
@@ -1,7 +1,13 @@
 Flipper.configure do |config|
   config.default do
-    # pick an adapter, this uses memory, any will do
-    adapter = Flipper::Adapters::ActiveRecord.new
+    adapter = if (cache_duration = ENV["CACHE_FEATURE_FLAGS_SECONDS"])
```

Comment:
nitpick: we have a convention of reading from ApplicationConfig, rather than from ENV directly - the initializers are executed alphabetically so 0_application_config.rb is loaded early enough.

Should this be set in the example .env_sample file as well (i.e. is it expected to be turned on in dev and test, or only used in production?)

Reply:
@djuber 

> we have a convention of reading from ApplicationConfig, rather than from ENV directly

Kinda.

```
➜  forem git:(cache-feature-flag-checks) grep -r ENV config app | wc -l
     122
➜  forem git:(cache-feature-flag-checks) grep -r ApplicationConfig config app | wc -l
     111
```

I'm sure using `ApplicationConfig` is preferred for the same reason that `FeatureFlag` is preferred (they both appear to be pass-through abstractions), but due to this ratio it's not worth reopening my text editor to change it for this one occurrence. I'll leave it as-is for now and someone can go in and change it later when they change all of the rest, probably with a `sed` one-liner and/or a linter autocorrect rule.

> Should this be set in the example .env_sample file as well (i.e. is it expected to be turned on in dev and test, or only used in production?)

It should probably only be used in production, but I didn't do a `Rails.env` check specifically because I think those make things impossible to test/debug without changing the code, which makes me want to avoid testing/debugging them. 😄 

I also don't think it makes sense for most Forems. It's an optimization that makes sense for us at our volume, but I doubt most Forem instances have enough traffic for this query to take any significant time. DEV executes this query for ~105ms of every second of the day on average, but I'd be surprised if most Forem instances were spending that much time on entire HTTP requests.

Reply:
> I'm sure using `ApplicationConfig` is preferred for the same reason that `FeatureFlag` is preferred (they both appear to be pass-through abstractions

Exactly this. Both wrappers were added so we don't commit to a particular implementation and later have to go change a bunch of call sites if we e.g. decide to swap Flipper for another gem.